src/fairchem/core/common/utils.py

```diff
@@ -1370,3 +1372,18 @@ def get_loss_module(loss_name):
         raise NotImplementedError(f"Unknown loss function name: {loss_name}")
 
     return loss_fn
+
+
+def load_model_and_weights_from_checkpoint(checkpoint_path: str) -> nn.Module:
+    if not os.path.isfile(checkpoint_path):
+        raise FileNotFoundError(
+            errno.ENOENT, "Checkpoint file not found", checkpoint_path
+        )
+    logging.info(f"Loading checkpoint from: {checkpoint_path}")
+    checkpoint = torch.load(checkpoint_path)
```

Comment:
Do we want to use map_location here? in case we are loading on CPU or GPU?

Reply:
good point